Kernel launch fails with "Disconnected from sync task" — broken pipe during streaming load

## Summary

`start_kernel()` fails for every notebook (fresh or existing, any `env_source`) from the Python bindings. The daemon is healthy — `ping()`, `status()`, `get_cells()` all work. Only kernel launch fails.

The Tauri frontend also cannot start kernels for saved notebooks (e.g. `notebooks/chaos.ipynb`), though the failure is silent — no error surfaces to the UI or daemon log.

## Error

```
runtimed.RuntimedError: Disconnected from sync task
```

## Root Cause (partial)

The daemon log shows:

```
[WARN] runtimed::notebook_sync_server: [notebook-sync] Streaming load failed for .../chaos.ipynb: Failed to send sync message: Broken pipe (os error 32)
[ERROR] runtimed::daemon: [runtimed] Connection error: Streaming load failed: Failed to send sync message: Broken pipe (os error 32)
```

The sync task between the Python client and the daemon dies during the streaming load phase. The `DocHandle`'s command channel closes, and all subsequent requests (including `LaunchKernel`) get `SyncError::Disconnected`.

**Key observation**: `get_cells()` succeeds before `start_kernel()` fails. The sync task is alive for read operations but dies when `LaunchKernel` is sent. Either:
1. The `LaunchKernel` request triggers a code path that crashes the sync task
2. The sync task has already died by the time `LaunchKernel` arrives, but the channel buffer masks it for earlier reads
3. There's a race condition in the streaming load that the kernel launch exposes

## Reproduction

```python
import runtimed

client = runtimed.Client()

# Fresh notebook — NOT pyproject-related
session = client.create_notebook(runtime='python')
print(session.connection_info)  # works
cells = session.get_cells()     # works (1 cell)

session.start_kernel(kernel_type='python', env_source='uv:prewarmed')
# RuntimedError: Disconnected from sync task
```

Every `env_source` fails: `auto`, `uv:prewarmed`, `uv:pyproject`.

The daemon is healthy throughout:
```python
client.ping()    # True
client.status()  # {'uv_available': 5, 'conda_available': 3, ...}
```

## Visibility Gap

- The **daemon log** only shows `WARN`-level entries. The kernel launch attempt from the Tauri frontend (visible in Tauri logs as `launch_kernel_via_daemon: type=auto, env_source=auto`) does NOT appear in the daemon log at all.
- The **Tauri app** silently fails — the user clicks Run, nothing happens, no error shown.
- The **Python bindings** at least surface the error, but "Disconnected from sync task" doesn't indicate what actually went wrong.

## Context

Discovered during gremlin stress testing (#990). Initially thought to be related to the pyproject.toml workspace root move, but reproduces on fresh UUID-based notebooks with `uv:prewarmed` — so it's not about pyproject detection.

May be pre-existing on `main` — the workspace move surfaced it because `notebooks/chaos.ipynb` now finds the repo root `pyproject.toml`, but the underlying broken pipe issue affects all notebooks.

## Investigation Leads

- `notebook_sync::connect::connect()` creates the `DocHandle` with a background sync task
- `DocHandle::send_request()` uses `cmd_tx` channel → `SyncError::Disconnected` when the task is dead
- The daemon's `streaming_load_cells()` fails with broken pipe — the client-side socket may be closing prematurely
- `session_core::connect_with_socket()` stores the handle but doesn't monitor the sync task health

## Files

| File | Role |
|------|------|
| `crates/runtimed-py/src/session_core.rs` | `connect_with_socket()`, `start_kernel()` |
| `crates/notebook-sync/src/handle.rs` | `DocHandle::send_request()`, `SyncError::Disconnected` |
| `crates/notebook-sync/src/error.rs` | `SyncError::Disconnected` definition |
| `crates/runtimed/src/notebook_sync_server.rs` | `streaming_load_cells()`, `handle_notebook_request(LaunchKernel)` |

--- a/crates/runtimed-py/src/session.rs
+++ b/crates/runtimed-py/src/session.rs
@@ -67,9 +67,11 @@ impl Session {
         ))?;
 
         state.peer_label = peer_label.clone();
-        drop(runtime);
 
-        Self::from_state(notebook_id, state, peer_label)
+        // Keep the runtime alive — the sync task was spawned on it during
+        // connect_open. Dropping the runtime would cancel the sync task and
+        // break all subsequent daemon requests (start_kernel, execute, etc.).
+        Self::from_state_with_runtime(runtime, notebook_id, state, peer_label)
     }
 
     /// Create a notebook without deprecation warning (used by Client).
@@ -95,11 +97,11 @@ impl Session {
             }
         }
 
-        let rt = Runtime::new().map_err(to_py_err)?;
+        let runtime = Runtime::new().map_err(to_py_err)?;
         let working_dir_buf = working_dir.map(PathBuf::from);
         let actor_label = peer_label.as_deref().map(session_core::make_actor_label);
 
-        let (notebook_id, mut state, _info) = rt.block_on(session_core::connect_create(
+        let (notebook_id, mut state, _info) = runtime.block_on(session_core::connect_create(
             socket_path,
             runtime_type,
             working_dir_buf,
@@ -107,9 +109,8 @@ impl Session {
         ))?;
 
         state.peer_label = peer_label.clone();
-        drop(rt);
 
-        Self::from_state(notebook_id, state, peer_label)
+        Self::from_state_with_runtime(runtime, notebook_id, state, peer_label)
     }
 
     /// Join an existing notebook room by ID (used by Client).
@@ -124,9 +125,9 @@ impl Session {
         state.peer_label = peer_label.clone();
         state.actor_label = actor_label;
 
-        let rt = Runtime::new().map_err(to_py_err)?;
+        let runtime = Runtime::new().map_err(to_py_err)?;
         let state_arc = Arc::new(Mutex::new(state));
-        rt.block_on(session_core::connect_with_socket(
+        runtime.block_on(session_core::connect_with_socket(
             &state_arc,
             notebook_id,
             socket_path,
@@ -136,18 +137,22 @@ impl Session {
             .map_err(|_| to_py_err("Failed to unwrap session state"))?
             .into_inner();
 
-        drop(rt);
-        Self::from_state(notebook_id.to_string(), state, peer_label)
+        Self::from_state_with_runtime(runtime, notebook_id.to_string(), state, peer_label)
     }
 
     /// Create a pre-connected Session from a notebook_id and SessionState.
     /// Used by Client.open_notebook() / Client.create_notebook() / Client.join_notebook().
-    pub(crate) fn from_state(
+    /// Create a Session with a provided runtime.
+    ///
+    /// The runtime MUST be the same one that the sync task was spawned on
+    /// during connect. Dropping that runtime kills the sync task and breaks
+    /// all daemon communication.
+    pub(crate) fn from_state_with_runtime(
+        runtime: Runtime,
         notebook_id: String,
         state: SessionState,
         peer_label: Option<String>,
     ) -> PyResult<Self> {
-        let runtime = Runtime::new().map_err(to_py_err)?;
         let override_arc = Arc::new(std::sync::Mutex::new(None));
         if let Some(ref rx) = state.broadcast_rx {
             session_core::spawn_rekey_watcher(rx, Arc::clone(&override_arc), runtime.handle());

